packaging/windows/install_script.nsis

```diff
@@ -0,0 +1,237 @@
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "DisplayName" "${APPNAME} - ${DESCRIPTION}"
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "UninstallString" "$\"$INSTDIR\uninstall.exe$\""
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "QuietUninstallString" "$\"$INSTDIR\uninstall.exe$\" /S"
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "InstallLocation" "$\"$INSTDIR$\""
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "DisplayIcon" "$\"$INSTDIR\logo.ico$\""
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "Publisher" "$\"https://github.com/grafana/agent$\""
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "HelpLink" "$\"${HELPURL}$\""
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "URLUpdateInfo" "$\"${UPDATEURL}$\""
+    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "URLInfoAbout" "$\"${ABOUTURL}$\""
+    # There is no option for modifying or repairing the install
+    WriteRegDWORD HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "NoModify" 1
+    WriteRegDWORD HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${APPNAME}" "NoRepair" 1
+
+    # Write the config file, its easier to do this way than replacing an values in a templated file
+
+    FileOpen $9 "$INSTDIR\agent-config.yaml" w ;Opens a Empty File and fills it
+    FileWrite $9 `server:$\n`;
+    FileWrite $9 `  http_listen_port: 12345$\n`
+    FileWrite $9 `prometheus:$\n`
+    FileWrite $9 `  wal_directory: /tmp/grafana-agent-wal$\n`;
+    FileWrite $9 `  global:$\n`
+    FileWrite $9 `    scrape_interval: 15s$\n`
+    FileWrite $9 `  configs:$\n`
+    FileWrite $9 `    - name: integrations$\n`
+    FileWrite $9 `      scrape_configs:$\n`
+    ${If} $EnableExporterValue == ${BST_CHECKED}
+        FileWrite $9 `        - job_name: integrations/windows-exporter$\n`
+        FileWrite $9 `          metrics_path: /metrics$\n`
+        FileWrite $9 `          static_configs:$\n`
+        FileWrite $9 `            - targets:$\n`
+        FileWrite $9 `                - localhost:9182$\n`
+        FileWrite $9 `      remote_write:$\n`
+        FileWrite $9 `        - url: $RemoteUrl$\n`
+        FileWrite $9 `          basic_auth:$\n`
+        FileWrite $9 `            username: "$RemoteUser"$\n`
+        FileWrite $9 `            password: $RemotePassword$\n`
+    ${EndIf}
+    FileWrite $9 `integrations:$\n`
+    FileWrite $9 `  prometheus_remote_write:$\n`
+    FileWrite $9 `    - url: $RemoteUrl$\n`
+    FileWrite $9 `      basic_auth:$\n`
+    FileWrite $9 `        username: "$RemoteUser"$\n`
+    FileWrite $9 `        password: $RemotePassword`
+    FileClose $9 ; and close the file
```

Comment:
This looks like it'll be hard to maintain. Is there any other way we can do this?

Reply:
The replace in file functionality in NSIS looked even more cumbersome, in my mind this is more of a starter template to get going with the just the windows exporter, then the end user would modify it to be whatever meets there need. I cna brainstorm some more on this though.

Reply:
can we run arbitrary code during the installer? we can write a custom agentctl command for doing templating, which IMO should also probably be installed to give the user WAL utilities

Reply:
Yes, we can run an executable, will also bundle the ctl.

Reply:
👍 IMO let's just bundle in agentctl but leave the config file as is for now. We can revisit later.